src/Form/ReactionRuleAddForm.php

```diff
@@ -81,10 +91,36 @@ public function form(array $form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function save(array $form, FormStateInterface $form_state) {
+    $event_name = $form_state->getValue('events')[0]['event_name'];
+    if ($handler = $this->getEventHandler($event_name)) {
+      $handler->extractConfigurationFormValues($form['event_configuration'], $form_state);
+      $this->entity->set('configuration', $handler->getConfiguration());
+      $this->entity->set('events', [['event_name' => $event_name . '--' . $handler->getConfiguration()['bundle']]]);
+    }
     parent::save($form, $form_state);
 
     drupal_set_message($this->t('Reaction rule %label has been created.', ['%label' => $this->entity->label()]));
     $form_state->setRedirect('entity.rules_reaction_rule.edit_form', ['rules_reaction_rule' => $this->entity->id()]);
   }
 
+  /**
+   * Gets event handler class.
+   *
+   * @param $event_name
+   *   The event base name.
+   * @param array $configuration
+   *   The event configuration.
+   *
+   * @return \Drupal\rules\Core\RulesConfigurableEventHandlerInterface|null
+   *   The event handler, null if there is no proper handler.
+   */
+  protected function getEventHandler($event_name, $configuration = []) {
```

Comment:
seems fine. but it should explain that there is only a handler when the event is configurable